spock-core/src/main/java/org/spockframework/runtime/Condition.java

```diff
@@ -115,8 +119,23 @@ private void createRendering() {
     rendering = builder.toString();
   }
 
-  private String flatten(String text) {
-    return pattern.matcher(text).replaceAll(" ");
+  private static CharSequence stripAndFlatten(String text) {
+    Matcher m = pattern.matcher(text);
+    StringBuffer sb = new StringBuffer();
```

Comment:
Ok, if the Matcher API requires the `StringBuffer` then it makes sense to use it.